File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int i=0,j=1;
        vector<int>ans(nums.size());
        for(int n:nums){
            if(n>0){
                ans[i]=n;
                i+=2;
            }else{
                ans[j]=n;
                j+=2;
            }
        }
        return ans;


    }
};

int crown =[](){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return 0;
}();

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.go
func rearrangeArray(nums []int) []int {
    n := len(nums)
    ans := make([]int, n)
    posIndex := 0
    negIndex := 1
    
    for _, num := range nums {
        if num > 0 {
            ans[posIndex] = num
            posIndex += 2
        }
    }
    
    for _, num := range nums {
        if num < 0 {
            ans[negIndex] = num
            negIndex += 2
        }
    }

    return ans
}
